### PR TITLE
drivers: serial: nrf_sw_lpuart: Add support for nrf54l15dk to low power UART

### PR DESCRIPTION
**Pull request by nordic-krch**
Update low power UART driver to support nrf54l15dk. 
Two modifications added to the driver:
- Finalized support for multi instance GPIOTE
- Add check if REQ and RDY pins are not used from GPIO port which does not work with GPIOTE (P2 in case of nrf54l15)
- Use `SYS_FOREVER` timeout in `uart_rx_enable` call to the UART device used by the low power UART.

Added support for nrf54l15dk//cpuapp in the sample and test.
Additional minor cleanup in the sample and test.

### CODE REVIEW

**Comment by nika-nordic** [comment]
@nordic-krch don't we need backport-label so this PR is added to NCS 2.8.0 release?

**Comment by nordic-krch** [comment]
Added backport label.

**Review by nika-nordic — APPROVED** [review]

**Comment by nordic-krch** [comment]
@anangl can you take a look?

File: samples/peripheral/lpuart/sample.yaml [hunk 1]
```diff
@@ -23,23 +23,24 @@ tests:
...
     platform_exclude: native_posix
     tags: sysbuild ci_samples_peripheral_lpuart
-  sample.peripheral.lpuart_nrf52840_debug:
+  sample.peripheral.lpuart.debug:
     sysbuild: true
     build_only: true
-    platform_allow: nrf52840dk/nrf52840
-    integration_platforms:
+    platform_allow:
+      - nrf52833dk/nrf52833
```
Comment (anangl) [inline samples/peripheral/lpuart/sample.yaml:30]: Was `nrf52dk/nrf52832` omitted here intentionally?

**Review by anangl — APPROVED** [review]

**Pull request merged**